packages/patrol_cli/lib/src/compatibility_checker.dart

```diff
@@ -0,0 +1,119 @@
+import 'dart:async';
+
+import 'package:dispose_scope/dispose_scope.dart';
+import 'package:file/file.dart';
+import 'package:patrol_cli/src/base/constants.dart';
+import 'package:patrol_cli/src/base/exceptions.dart';
+import 'package:patrol_cli/src/base/process.dart';
+import 'package:process/process.dart';
+import 'package:version/version.dart';
+
+class CompatibilityChecker {
+  CompatibilityChecker({
+    required Directory projectRoot,
+    required ProcessManager processManager,
+  })  : _projectRoot = projectRoot,
+        _processManager = processManager,
+        _disposeScope = DisposeScope();
+
+  final Directory _projectRoot;
+  final ProcessManager _processManager;
+  final DisposeScope _disposeScope;
+
+  Future<void> checkVersionsCompatibility() async {
+    String? packageVersion;
+    final completer = Completer<String?>();
+
+    await _disposeScope.run((scope) async {
+      final process = await _processManager.start(
+        ['flutter', 'pub', 'deps', '--style=list'],
```

Comment:
add `--no-version-check`, otherwise it's gonna fail when Flutter prints its usual "New Flutter version available!" message

Reply:
example: https://github.com/leancodepl/patrol/blob/e7c7fed2672668c39fc22e6d0f96300bba2baeee/packages/patrol_cli/lib/src/crossplatform/flutter_tool.dart#L91-L104

Reply:
We should also add `--suppress-analytics` to **every** Flutter CLI invocation we do. Currently JSON parsing fails when Flutter prints that message.